apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/LogReportServiceClient.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.remote;
+
+import java.util.List;
+
+import org.apache.skywalking.apm.agent.core.boot.BootService;
+import org.apache.skywalking.apm.agent.core.boot.DefaultImplementor;
+import org.apache.skywalking.apm.commons.datacarrier.consumer.IConsumer;
+import org.apache.skywalking.apm.network.logging.v3.LogData;
+
+@DefaultImplementor
+public class LogReportServiceClient implements BootService, IConsumer<LogData> {
+
+    @Override
+    public void prepare() throws Throwable {
+
+    }
+
+    @Override
+    public void boot() throws Throwable {
+
+    }
+
+    @Override
+    public void onComplete() throws Throwable {
+
+    }
+
+    @Override
+    public void shutdown() throws Throwable {
+
+    }
+
+    @Override
+    public void init() {
+
+    }
+
+    public void produce(LogData logData) {
+
+    }
```

Comment:
Why this?

Reply:
`GRPCLogReportServiceClient` needs to produce `LogData` to `DataCarrier` and consume it by sending it to the gRPC server. In the `apm-toolkit-activation` module, `GRPCLogClientAppender` produce `LogData` by `LogReportServiceClient`, so add this method in here and  implement the interface `IConsumer<LogData>`.

Reply:
I know you need to. My question is, why need this in core default empty service?

Reply:
@wu-sheng  In toolkit module, only LogReportServiceClient can be obtained by `ServiceManager.INSTANCE.findService()`. There is a need for class LogReportServiceClient to provide a method for producing data, so I add this default empty service. Is there something wrong with the way I use it?

Reply:
My thought is, you should be able to `findService` and cast to your `override` implementation.

Reply:
I tried it by using `findService(LogReportServiceClient.class)` to obtain the client and then cast it to `GRPCLogReportServiceClient`, but  it occurs an error  like `GRPCLogReportServiceClient cannot be cast to GRPCLogReportServiceClient`. The way doesn't seem to work.

Reply:
I mean remove the interface, your previous error is caused by `LogReportServiceClient` doesn't exist.